packages/shared.net/src/createClientContext.ts

```diff
@@ -82,8 +82,9 @@ export function createClientContext<
 
   return {
     ontology,
-    stack,
+    stack: baseUrl,
+    baseUrl,
     fetch: fetchWrapper,
-    tokenProvider,
+    tokenProvider: async () => await tokenProvider(),
```

Comment:
Ensure it always returns a promise without breaking the api contract of what we take.